src/main/java/com/google/devtools/build/lib/analysis/starlark/StarlarkCustomCommandLine.java

```diff
@@ -459,7 +478,24 @@ private int addToFingerprint(
         starlarkSemantics = (StarlarkSemantics) arguments.get(argi++);
       }
 
+      // NestedSets and lists never result in the same fingerprint as the former prepends the order.
```

Comment:
I didn't follow this comment at first - consider clarifying that this happens in the `addNestedSetToFingerprint` call.